sam/cigar_test.go

```diff
@@ -0,0 +1,32 @@
+// Copyright ©2013 The bíogo Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package sam
+
+import "gopkg.in/check.v1"
+
+type CigarSuite struct{}
+
+var _ = check.Suite(&CigarSuite{})
+
+var cigTests = []struct {
+	cig  []byte
+	ref  int
+	read int
+}{
+	{[]byte("151M"), 151, 151},
+	{[]byte("10S10M"), 10, 20},
+	{[]byte("11H11M"), 11, 11},
+	{[]byte("11H1D11M"), 12, 11},
```

Comment:
Last thing. It's probably worth having a test for a skipped (N) operation, and I think just put this test in sam_test.go